.github/workflows/unit_test_4gpu.yaml

```diff
@@ -29,5 +29,7 @@ jobs:
         conda activate "${CONDA_ENV}"
 
         python -m pip install --force-reinstall --pre torch --index-url https://download.pytorch.org/whl/nightly/cu121
+        python -m pip install --pre torchdata --index-url https://download.pytorch.org/whl/nightly/
+        pip install -e ".[dev]"
```

Comment:
Do we need this?

Reply:
Answered below though I am not sure what tests are run for the 4gpu case

Reply:
ditto: Let's remove this `pip install -e ".[dev]"` if not necessary